From: Anthony Liguori <aliguori@us.ibm.com>
Subject: SLES9 installation: answer MULTISESSION cdrom ioctl

Add an old hack from 2.4 Xen to enable at least some response to 
MULTISESSION ioctls (otherwise isofs won't mount the device

Acked-by: garloff@suse.de

diff -uNrp linux-2.6.10.xen2/drivers/xen/blkfront/blkfront.c linux-2.6.10.xen3/drivers/xen/blkfront/blkfront.c
--- linux-2.6.10.xen2/drivers/xen/blkfront/blkfront.c	2005-01-21 19:45:14.635717715 +0100
+++ linux-2.6.10.xen3/drivers/xen/blkfront/blkfront.c	2005-01-21 19:55:01.545314512 +0100
@@ -223,6 +223,7 @@ int blkif_release(struct inode *inode, s
 int blkif_ioctl(struct inode *inode, struct file *filep,
                 unsigned command, unsigned long argument)
 {
+	int i;
     /*  struct gendisk *gd = inode->i_bdev->bd_disk; */
 
     DPRINTK_IOCTL("command: 0x%x, argument: 0x%lx, dev: 0x%04x\n",
@@ -234,6 +235,12 @@ int blkif_ioctl(struct inode *inode, str
         /* return ENOSYS to use defaults */
         return -ENOSYS;
 
+    case CDROMMULTISESSION:
+        DPRINTK("FIXME: support multisession CDs later\n");
+        for ( i = 0; i < sizeof(struct cdrom_multisession); i++ )
+            if ( put_user(0, (byte *)(argument + i)) ) return -EFAULT;
+        return 0;
+
     default:
         printk(KERN_ALERT "ioctl %08x not supported by Xen blkdev\n",
                command);
